juju/juju
# Move RetryProvisioning and ProvisioningScript methods off client facade

A number of methods exit on the legacy client facade that need to be moved off to bespoke facades. This PR does the machine provisioning related ones. The main apis to move off are `RetryProvisioning` and `ProvisioningScript`. In addition, the `add-machine` CLI used really old machine calls dating back from 2017 and Juju 2.2. These calls had already been moved to the machine manager facade and used elsewhere. So tidy up was done here also.

The upshot is that the 2.9 client will prefer the moved methods on the new machine manager v7 facade and fall back to those on the client facade for older controllers. The apiserver backend methods are kept to server older clients and will be removed in juju 3.

There's some followup work to do still because the code moved across from the client facade to the machine manager facade relies on older test mocks. These have been kept in tact for convenience.

Also, some really old, unused client api Public/Private address methods are removed. The apiserver code will be removed in juju 3.

Note that with this change, the juju CLI for add/remove machine may break when run against older 2.2 or earlier controllers, but I doubt we care about that anymore.

## QA steps

on a 2.9.31 controller and also a controller bootstrapped off this PR:

add a machine and also a manual machine and remove them
run retry-provisioning - it will fail saying the machine is not in error, that that indicates the api call has been made as expected
